### PR TITLE
Use `package:patrol` for Android e2e tests.

### PR DESCRIPTION
**Pull request by Jonas-Sander**
Use [`package:patrol`](https://pub.dev/packages/patrol) for integration/e2e tests for Android.   

The biggest benefit is that each test can run from a completely clean state, which is not possible with the `integration_test` package. Because of this we shouldn't have the issues with e.g. the auth state being kept for each test (or logging out and back in which breaks stuff).

For other platforms I kept our old tests (renamed to `integration_test/integration_test_old.dart`):
* Web should theoretically work as long as no native interaction is used, but I couldn't set it up practically. There is an [open issue for web ](https://github.com/leancodepl/patrol/issues/733) in their repo.
* iOS should work, but I couldn't get it running.
* macOS is never mentioned, so I don't know if it's possible to run tests for mac.

From https://patrol.leancode.co/:
> Patrol lets you [access native features of the platform](https://patrol.leancode.co/native/overview) that the Flutter app is running on. Finally, you can interact with permission dialogs, notifications, WebViews, change device settings, toggle Wi-Fi, and much more – and you can code this very easily in plain Dart.
> 
> Patrol also provides a [new custom finder system](https://patrol.leancode.co/finders/overview) that extends Flutter's default finders, making them shorter and easier to understand. Patrol's custom finders, coupled with [Hot Restart](https://patrol.leancode.co/cli-commands/develop), make writing integration tests dramatically faster, easier and more fun!

Command to running the tests:
```
patrol build android \
 --flavor prod \
 --dart-define USER_1_EMAIL=example@sharezone.net \
 --dart-define USER_1_PASSWORD=foobar \
 -t integration_test/app_test.dart
```

### CODE REVIEW

**Comment by Jonas-Sander** [comment]
@nilsreichardt Ready for review.

I couldn't get iOS set up, so for now I left it as is.
Maybe you could try your luck sometime in the future with setting it up.

Also the macOS job seems to be making problems, do you know how it can be fixed?

**Review by nilsreichardt — APPROVED** [review]
LGTM

**Comment by nilsreichardt** [comment]
> iOS should work, but I couldn't get it running.

Even, if we can get Patrol running for iOS, we can't re-use the tests for Android because on iOS you can't clear the app state after each test. Because of our sign-out bug, we still need on iOS one large e2e test.

File: .github/workflows/integration_tests_app_ci.yml [hunk 1]
```diff
@@ -99,18 +99,10 @@ jobs:
...
-        env:
-          # In gradlew, we can't just pass --dart-define as we do with "flutter
-          # test". We need to convert the key and the value into base64, like
-          # "USER_1_EMAIL=user@example.com" becomes to
-          # "VVNFUl8xX0VNQUlMPXVzZXJAZXhhbXBsZS5jb20=".
-          #
-          # The secrets already contain the base64 encoded values. We don't
-          # encode them in the workflow file, because we could easily leak the
-          # base64 encoded version in logs.
-          USER_1_EMAIL_BASE64: ${{ secrets.INTEGRATION_TEST_USER_1_EMAIL_DART_DEFINE_BASE64 }}
```
[the remark this answers is not in the document]
Comment (Jonas-Sander) [reply .github/workflows/integration_tests_app_ci.yml:112]: I think so, yes

**Pull request merged**